### PR TITLE
Set up firebase authentication

### PR DESCRIPTION
**Pull request by lin8231**
* Auth flow now working properly with firebase auth 
    * **TO RUN THIS AFTER PULLING PLEASE DO:**
    * Get the config/firebaseConfig.js file from mercurylin@ and paste this into the services/ directory
    * Make sure that this file is gitignored (should already be after pulling)
* User ID is queried from backend by passing in email after user is logged in 
    * User ID is later used to query the database for the user's other information
    * No backend support for querying for ID using email as of yet, so **ID is currently hardcoded to 123**

### CODE REVIEW

File: project/frontend/src/App.js [hunk 1]
```diff
@@ -1,67 +1,68 @@
...
+    handleAuth: signInWithGoogle,
+    clearAuth: signOut,
   };
 
+  let pageRoute = !user.signedIn ? (
+    <Switch>
+      <Route path="/signin" component={LandingPage}></Route>
+      <Redirect to="/signin"></Redirect>
+    </Switch>
+  ) : compId == 0 ? (
```
Comment (texm) [inline project/frontend/src/App.js:51]: A comment explaining what this logic is for would be nice, or maybe a const value like (NO_COMPETITION=0) in place of just a 0.

File: project/frontend/package.json [hunk 2]
```diff
@@ -7,6 +7,8 @@
     "@testing-library/react": "^11.2.2",
     "@testing-library/user-event": "^12.6.0",
     "classnames": "^2.2.6",
+    "dotenv": "^8.2.0",
```
Comment (texm) [inline project/frontend/package.json:10]: I don't think we need this dep anymore?

**Review by texm — APPROVED** [review]
LGTM, with nitpicks

File: project/frontend/src/App.js (hunk 1 above)
Comment (lin8231) [reply project/frontend/src/App.js:51]: Good point, good idea. Fix coming right up

File: project/frontend/package.json (hunk 2 above)
Comment (lin8231) [reply project/frontend/package.json:10]: Forgot to remove - Will do now.

**Review by texm — APPROVED** [review]

**Review by roonil-wazlib — APPROVED** [review]
LGTM

**Pull request merged**